docs/access_and_permissions.md

```diff
@@ -244,6 +299,61 @@ flowchart TD
     G --Pundit.policy_scope(current_user, SupportedPermission)--> H("SupportedPermissionPolicy (scope)")
 ```
 
+#### Users signin permissions create
+
+These dependencies determine whether a user can:
+
+- complete the controller action
+- grant a given user access to an app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#create)
+    --authorize [application, user: @user], policy_class: Users::ApplicationPolicy
+    --> B(Users::ApplicationPolicy#grant_signin_permission?)
+
+    A --"UserUpdate.new(current_user, { supported_permission_ids: }, current_user, user_ip_address).call"
+    --> C(UserUpdate#call)
+
+    C --SupportedPermissionParameterFilter.new(current_user, user, user_params) [...] .filtered_supported_permission_ids
+    --> D(SupportedPermissionParameterFilter#filtered_supported_permission_ids)
+    D --Pundit.policy_scope(current_user, SupportedPermission)--> E("SupportedPermissionPolicy (scope)")
+```
+
+#### Users signin permissions delete
+
+These dependencies determine whether a user can:
+
+- view the confirmation screen for revoking access to a given app
+
+```mermaid
+flowchart TD
+    A(Users::SigninPermissionsController#delete)
+    --authorize [application, user: @user], policy_class: Users::ApplicationPolicy
```

Comment:
`authorize [{ application: @application, user: @user }], :remove_signin_permission?, policy_class: Users::ApplicationPolicy`?

Reply:
It's not using an instance variable here, but I wasn't sure how we might shorthand this so I don't need to add another section of the diagram here - I was using `application` here as the "found application" in the line above, do you think an instance variable would make this clearer?

Reply:
Would `application:` be enough, do you think?

Reply:
I'm not sure I understand what you mean, but it looks like it's using an instance variable and a named policy method here: https://github.com/alphagov/signon/blob/main/app/controllers/users/signin_permissions_controller.rb#L17

Reply:
Are you looking at a different action?

Reply:
Sorry, getting confused about all of these, must have been looking at the wrong action